### PR TITLE
Staking cleanups

### PR DESCRIPTION
**Pull request by 0xAshish**
Few cleanups for stakeManager
- refactored sig check function so that external users can check 2/3 power sigs, and can be used with things like `only2/3Valiadtors`.
- added checkpoint minimum block interval limit to rate-limit the checkpoints with rewards.
- Auction period is fixed now, automatically closes auction period window.

### CODE REVIEW

File: contracts/staking/StakeManager.sol [hunk 1]
```diff
@@ -139,7 +143,13 @@ contract StakeManager is Validator, IStakeManager, RootChainable, Lockable {
 
   function startAuction(uint256 validatorId, uint256 amount) external {
     require(isValidator(validatorId));
+    // when dynasty period is updated validators are in cool down period
+    require(replacementCoolDown == 0 || replacementCoolDown <= currentEpoch, "Cool down period");
     require(auctionPeriod >= currentEpoch.sub(validatorAuction[validatorId].startEpoch), "Invalid auction period");
+    // dynasty--auctionPeriod--dynasty--auctionPeriod--dynasty
+    // make sure that its `auctionPeriod` window
+    require((currentEpoch.sub(validators[validatorId].activationEpoch) % dynasty.add(auctionPeriod)) > dynasty, "Not an auction time");
```
Comment (atvanguard) [inline contracts/staking/StakeManager.sol]: A comment saying what this formula means might be a good idea.

Comment (0xAshish) [reply contracts/staking/StakeManager.sol]: yeah makes sense will add.

Comment (ptsayli) [reply contracts/staking/StakeManager.sol]: i am sayli

File: contracts/staking/StakeManager.sol [hunk 2]
```diff
@@ -479,9 +499,28 @@ contract StakeManager is Validator, IStakeManager, RootChainable, Lockable {
...
+  function checkSignatures(uint256 blockInterval, bytes32 voteHash, bytes32 stateRoot, bytes memory sigs) public onlyRootChain returns(uint256) {
+    uint256 stakePower;
+    uint256 _totalStake;
+    (stakePower, _totalStake) = checkTwoByThreeMajority(voteHash, sigs);
+    // checkpoint rewards are based on BlockInterval multiplied on `checkpointReward`
+    // with actual `blockInterval`
+    // eg. checkpointReward = 10 Tokens, checkPointBlockInterval = 250, blockInterval = 500 then reward
+    // for this checkpoint is 20 Tokens
+    uint256 _reward = checkPointBlockInterval.mul(checkpointReward).div(blockInterval);
+    _reward = Math.max(checkpointReward, _reward.mul(stakePower).div(_totalStake));
```
Comment (atvanguard) [inline contracts/staking/StakeManager.sol]: `.min`

**Pull request merged**